Mathlib/NumberTheory/ADEInequality.lean

```diff
@@ -0,0 +1,286 @@
+This solution is related to the Dynkin diagrams $E_7$. -/
+def E7 : Multiset ℕ+ :=
+  E' 4
+#align ADE_inequality.E7 ADEInequality.E7
+
+/-- `E8 := {2,3,5}` is a `multiset ℕ+`
+that is a solution to the inequality
+`(p⁻¹ + q⁻¹ + r⁻¹ : ℚ) > 1`.
+
+This solution is related to the Dynkin diagrams $E_8$. -/
+def E8 : Multiset ℕ+ :=
+  E' 5
+#align ADE_inequality.E8 ADEInequality.E8
+
+/-- `sum_inv pqr` for a `pqr : multiset ℕ+` is the sum of the inverses
+of the elements of `pqr`, as rational number.
+
+The intended argument is a multiset `{p,q,r}` of cardinality `3`. -/
+def sumInv (pqr : Multiset ℕ+) : ℚ :=
+  Multiset.sum (pqr.map fun (x : ℕ+) => x⁻¹)
+#align ADE_inequality.sum_inv ADEInequality.sumInv
+
+theorem sumInv_pqr (p q r : ℕ+) : sumInv {p, q, r} = (p : ℚ)⁻¹ + (q : ℚ)⁻¹ + (r : ℚ)⁻¹ := by
+  simp only [sumInv, add_zero, insert_eq_cons, add_assoc, map_cons, sum_cons,
+    map_singleton, sum_singleton]
+#align ADE_inequality.sum_inv_pqr ADEInequality.sumInv_pqr
+
+/-- A multiset `pqr` of positive natural numbers is `admissible`
+if it is equal to `A' q r`, or `D' r`, or one of `E6`, `E7`, or `E8`. -/
+def Admissible (pqr : Multiset ℕ+) : Prop :=
+  (∃ q r, A' q r = pqr) ∨ (∃ r, D' r = pqr) ∨ E' 3 = pqr ∨ E' 4 = pqr ∨ E' 5 = pqr
+#align ADE_inequality.admissible ADEInequality.Admissible
+
+theorem admissible_A' (q r : ℕ+) : Admissible (A' q r) :=
+  Or.inl ⟨q, r, rfl⟩
+#align ADE_inequality.admissible_A' ADEInequality.admissible_A'
+
+theorem admissible_D' (n : ℕ+) : Admissible (D' n) :=
+  Or.inr <| Or.inl ⟨n, rfl⟩
+#align ADE_inequality.admissible_D' ADEInequality.admissible_D'
+
+theorem admissible_E'3 : Admissible (E' 3) :=
+  Or.inr <| Or.inr <| Or.inl rfl
+#align ADE_inequality.admissible_E'3 ADEInequality.admissible_E'3
+
+theorem admissible_E'4 : Admissible (E' 4) :=
+  Or.inr <| Or.inr <| Or.inr <| Or.inl rfl
+#align ADE_inequality.admissible_E'4 ADEInequality.admissible_E'4
+
+theorem admissible_E'5 : Admissible (E' 5) :=
+  Or.inr <| Or.inr <| Or.inr <| Or.inr rfl
+#align ADE_inequality.admissible_E'5 ADEInequality.admissible_E'5
+
+theorem admissible_E6 : Admissible E6 :=
+  admissible_E'3
+#align ADE_inequality.admissible_E6 ADEInequality.admissible_E6
+
+theorem admissible_E7 : Admissible E7 :=
+  admissible_E'4
+#align ADE_inequality.admissible_E7 ADEInequality.admissible_E7
+
+theorem admissible_E8 : Admissible E8 :=
+  admissible_E'5
+#align ADE_inequality.admissible_E8 ADEInequality.admissible_E8
+
+theorem Admissible.one_lt_sumInv {pqr : Multiset ℕ+} : Admissible pqr → 1 < sumInv pqr := by
+  rw [Admissible]
+  rintro (⟨p', q', H⟩ | ⟨n, H⟩ | H | H | H)
+  · rw [← H, A', sumInv_pqr, add_assoc]
+    simp only [lt_add_iff_pos_right, PNat.one_coe, inv_one, Nat.cast_one]
+    apply add_pos <;> simp only [PNat.pos, Nat.cast_pos, inv_pos]
+  · rw [← H, D', sumInv_pqr]
+    conv_rhs => simp only [OfNat.ofNat, PNat.mk_coe]
+    norm_num
+  all_goals
+    rw [← H, E', sumInv_pqr]
+    conv_rhs => simp only [OfNat.ofNat, PNat.mk_coe]
```

Comment:
We can golf here by `conv_rhs => simp only [OfNat.ofNat, PNat.mk_coe]`.

Reply:
great!